tests/components/gogogate2/test_cover.py

```diff
@@ -0,0 +1,531 @@
+"""Tests for the GogoGate2 component."""
+from datetime import datetime, timedelta
+from unittest.mock import MagicMock, patch
+
+from gogogate2_api import GogoGate2Api
+from gogogate2_api.common import (
+    ActivateResponse,
+    ApiError,
+    Door,
+    DoorMode,
+    DoorStatus,
+    InfoResponse,
+    Network,
+    Outputs,
+    Wifi,
+)
+
+from homeassistant.components.cover import DOMAIN as COVER_DOMAIN
+from homeassistant.components.homeassistant import DOMAIN as HA_DOMAIN
+from homeassistant.const import (
+    CONF_IP_ADDRESS,
+    CONF_NAME,
+    CONF_PASSWORD,
+    CONF_PLATFORM,
+    CONF_USERNAME,
+    STATE_CLOSED,
+    STATE_CLOSING,
+    STATE_OPEN,
+    STATE_OPENING,
+    STATE_UNAVAILABLE,
+)
+from homeassistant.core import HomeAssistant
+
+from .common import ComponentFactory
+
+
+async def test_import_fail(
+    hass: HomeAssistant, component_factory: ComponentFactory
+) -> None:
+    """Test the failure to import."""
+    api = MagicMock(spec=GogoGate2Api)
+    api.info.side_effect = ApiError(22, "Error")
+
+    component_factory.api_class_mock.return_value = api
+
+    await component_factory.configure_component(
+        cover_config=[
+            {
+                CONF_PLATFORM: "gogogate2",
+                CONF_NAME: "cover0",
+                CONF_IP_ADDRESS: "127.0.1.0",
+                CONF_USERNAME: "user0",
+                CONF_PASSWORD: "password0",
+            }
+        ]
+    )
+
+    entity_ids = hass.states.async_entity_ids(COVER_DOMAIN)
+    assert not entity_ids
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5535891)